core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -565,12 +569,40 @@ class PartKeyLuceneIndex(ref: DatasetRef,
                            partId: Int, startTime: Long, endTime: Long): Document = {
     val partKeyBytesRef = new BytesRef(partKeyOnHeapBytes, partKeyBytesRefOffset, partKeyNumBytes)
     luceneDocument.get().reset(partId, partKeyBytesRef, startTime, endTime)
+
+    // If configured and enabled, Multi-column facets will be created on "partition-schema" columns
+    createMultiColumnFacets(partKeyOnHeapBytes, partKeyBytesRefOffset)
+
     cforRange { 0 until numPartColumns } { i =>
       indexers(i).fromPartKey(partKeyOnHeapBytes, bytesRefToUnsafeOffset(partKeyBytesRefOffset), partId)
     }
     luceneDocument.get().document
   }
 
+  // Multi-column facets to be created on "partition-schema" columns
+  private def createMultiColumnFacets(partKeyOnHeapBytes: Array[Byte], partKeyBytesRefOffset: Int): Unit = {
+    schema.options.multiColumnFacets.foreach(facetCols => {
+      facetCols match {
+        case (name, cols) => {
+          val concatFacetValue = cols.flatMap { col =>
+            val (columnInfo, pos) = schema.columnIdxLookup(col)
+            if (columnInfo.columnType == StringColumn) {
+              val base = partKeyOnHeapBytes
+              val offset = bytesRefToUnsafeOffset(partKeyBytesRefOffset)
+              val strOffset = schema.binSchema.blobOffset(base, offset, pos)
+              val numBytes = schema.binSchema.blobNumBytes(base, offset, pos)
+              Seq(new String(base, strOffset.toInt - UnsafeUtils.arayOffset,
```

Comment:
Why flatMap on a ``Seq`` of length 1 or 0, can we use ``map`` and either return a string, either empty or the value?

Reply:
I wanted to exclude the field altogether in case the field type is not `string`. But fixed it anyway 👍